src/character.cpp

```diff
@@ -5063,6 +5088,18 @@ void Character::update_needs( int rate_multiplier )
         mod_painkiller( -std::min( get_painkiller(), rate_multiplier ) );
     }
 
+    if( get_bp_effect_mod() > 0 ) {
+        modify_bp_effect_mod( -std::min( get_bp_effect_mod(), rate_multiplier ) );
```

Comment:
Another thing I'd like to point out is `bp_effect_mod`, `heartrate_effect_mod`, and `respiration_effect_mod` do indeed decrease slowly to 0 over time - but the problem here is, they don't seem to slowly increase to 0 when negative.
These mod values should also increase slowly to 0 when negative (for example, when blood pressure was decreased by an effect), this way, the player won't end up with permanently lower-than-usual values for these mods which would lead to potential problems in the future.
I would suggest adding another condition to check if these mods are negative, and if so, slowly increase them to 0 (basically, the reverse of what is done for the positive modifier values)

Reply:
You are correct. Nice find. I will fix this.

Reply:
Nice catch, fixed!